cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MainOutputCommitter.java

```diff
@@ -160,6 +160,7 @@ private void commitTx() throws IOException {
         LOG.warn("MapReduce Job transaction failed to commit");
         throw new TransactionFailureException("Failed to commit transaction " + transaction.getWritePointer());
       }
+      taskContext.postTxCommit();
```

Comment:
if this throws an exception, will that fail the job? I think the contract of postCommit() is that it should not cause failure, it is called as a "best effort". It would be strange, too, if the job failed, because the tx is already committed, and its writes will not be undone.